src/ai/storage.py

```diff
@@ -144,7 +144,9 @@ async def store_drone(message: discord.Message, message_copy=None):
         initiator = fetch_drone_with_drone_id(drone_id)
 
         # proceed if allowed, send error message if not
-        if initiator.id in [HIVE_MXTRESS_USER_ID, drone_to_store.id] + trusted_users:
+        if initiator.id in [drone_to_store.id] + trusted_users:
+            await initiate_drone_storage(drone_to_store, drone_id, target_id, time, purpose, message)
+        elif roles.has_role(message.author, roles.HIVE_MXTRESS):
             await initiate_drone_storage(drone_to_store, drone_id, target_id, time, purpose, message)
```

Comment:
as line 147 and 149 are the exact same, this can be reduced to

```suggestion
        if initiator.id in [drone_to_store.id] + trusted_users or roles.has_role(message.author, roles.HIVE_MXTRESS):
            await initiate_drone_storage(drone_to_store, drone_id, target_id, time, purpose, message)
```

add brackets to taste